src/storage/data_table.cpp

```diff
@@ -365,6 +365,7 @@ void DataTable::VerifyIndexBuffers() {
 	});
 }
 
+// Lock?
```

Comment:
Should not be required - the `CleanupAppend` in the version info (`RowVersionManager::CleanupAppend`) grabs fine-grained locks